gprofiler/spark/metrics.py

```diff
@@ -0,0 +1,409 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+# (C) Datadog, Inc. 2018-present. All rights reserved.
+# Licensed under a 3-clause BSD style license (see LICENSE.bsd3).
+#
+YARN_NODES_LABELS = ["node_hostname"]
+SPARK_APPS_LABELS = ["app_name", "app_id"]
+SPARK_APP_LABELS = ["app_name", "app_id"]
+SPARK_BATCH_LABELS = ["app_name", "app_id"]
+SPARK_STAGE_LABELS = ["app_name", "app_id", "status", "stage_id"]
+SPARK_TASK_SUMMARY_LABELS = ["app_name", "app_id", "status", "stage_id", "quantile"]
+SPARK_RDD_LABELS = ["app_name", "app_id"]
+
+YARN_CLUSTER_METRICS = {
+    "appsSubmitted": {"name": "yarn_cluster_appsSubmitted", "labels": []},
+    "appsCompleted": {"name": "yarn_cluster_appsCompleted", "labels": []},
+    "appsPending": {"name": "yarn_cluster_appsPending", "labels": []},
+    "appsRunning": {"name": "yarn_cluster_appsRunning", "labels": []},
+    "appsFailed": {"name": "yarn_cluster_appsFailed", "labels": []},
+    "appsKilled": {"name": "yarn_cluster_appsKilled", "labels": []},
+    "totalMB": {"name": "yarn_cluster_totalMB", "labels": []},
+    "availableMB": {"name": "yarn_cluster_availableMB", "labels": []},
+    "allocatedMB": {"name": "yarn_cluster_allocatedMB", "labels": []},
+    "availableVirtualCores": {
+        "name": "yarn_cluster_availableVirtualCores",
+        "labels": [],
+    },
+    "allocatedVirtualCores": {
+        "name": "yarn_cluster_allocatedVirtualCores",
+        "labels": [],
+    },
+    "totalNodes": {"name": "yarn_cluster_totalNodes", "labels": []},
+    "activeNodes": {"name": "yarn_cluster_activeNodes", "labels": []},
+    "lostNodes": {"name": "yarn_cluster_lostNodes", "labels": []},
+    "decommissioningNodes": {"name": "yarn_cluster_decommissioningNodes", "labels": []},
+    "decommissionedNodes": {"name": "yarn_cluster_decommissionedNodes", "labels": []},
+    "rebootedNodes": {"name": "yarn_cluster_rebootedNodes", "labels": []},
+    "shutdownNodes": {"name": "yarn_cluster_shutdownNodes", "labels": []},
+    "unhealthyNodes": {"name": "yarn_cluster_unhealthyNodes", "labels": []},
+    "containersAllocated": {"name": "yarn_cluster_containersAllocated", "labels": []},
+    "containersPending": {"name": "yarn_cluster_containersPending", "labels": []},
+}
+YARN_NODES_METRICS = {
```

Comment:
Consider - instead of manually adding `YARN_NODES_LABELS` to all metrics under it - add it automatically for all keys after the definition of the dict?

Reply:
Same for the other dicts which use labels, e.g `SPARK_APPLICATION_GAUGE_METRICS`